vortex-scalar/src/pvalue.rs

```diff
@@ -86,29 +107,6 @@ impl ToBytes for PValue {
     }
 }
 
-macro_rules! as_primitive {
```

Comment:
we had two variants of these, as_X and as_primitive<T>, I have kept the latter as it doesn't require macros and is a lot easier to customize for different type